teensy/teensy_main/teensy_main.ino

```diff
@@ -428,6 +391,14 @@ void pull_data(){
     set_sensor_status_flags(gps, imu, tpa);
 
     hdr = fill_in_hdr_array(hdr);
+    
+    // Will only sample once the system is initialized.
+    if (SEND_DATA)
```

Comment:
This is not ideal. Why do this check every time if `SEND_DATA` is only getting set to true in `setup`, and never changing again? See comment for more details.